training/distributed_training/pytorch/model_parallel/GPT2_huggingface_trainer/submit_smp_trainer.ipynb

```diff
@@ -0,0 +1,623 @@
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Set Up Hyperparameters, Metric Definitions, and MPI Options\n",
+    "The following `hyperparameters` dictionary is to pass arguments to the training script (`run_clm.py`) and set the model parallel configuration when creating the training job.\n",
+    "\n",
+    "Note that the `run_clm.py` file is currently modified to work with `SageMaker`, so if you want to run your own script, you'll need to add the relevant lines as seen in `run_clm.py`. You can find them quickly by searching for `SageMaker Support`.\n",
+    "\n",
+    "You can also add custom mpi flags. By default, we have `--mca btl_vader_single_copy_mechanism none` to remove unnecessary logs.\n",
+    "\n",
+    "Next we add a base metric definitions to enable the metric upload in SageMaker. You can add any further metric definitions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "save_steps = 60  # Set the interval for saving checkpoints\n",
+    "max_steps = 100  # Set the total number of steps you want to run\n",
+    "\n",
+    "hyperparameters = {\n",
+    "    \"output_dir\": \"/opt/ml/checkpoints\",\n",
+    "    \"overwrite_output_dir\": \"\",\n",
+    "    \"learning_rate\": 0.0002,\n",
+    "    \"do_train\": \"\",\n",
+    "    \"save_steps\": save_steps,\n",
+    "    \"max_steps\": max_steps,\n",
+    "    \"eval_steps\": 20,\n",
+    "    \"evaluation_strategy\": \"steps\",\n",
+    "    \"model_type\": \"gpt2\",\n",
+    "    \"tokenizer_name\": \"gpt2\",\n",
+    "    \"optim\": \"adamw_torch\",\n",
+    "    \"dataloader_drop_last\": True,\n",
+    "}\n",
+    "\n",
+    "\n",
+    "# The checkpoint path (hyperparameters['checkpoint-dir'] or checkpoint_s3_uri) is not unique per job.\n",
+    "# You need to modify as needed for different runs.\n",
+    "# If same path is used for unrelated runs, this may increase time when downloading unnecessary checkpoints,\n",
+    "# and cause conflicts when loading checkpoints.\n",
+    "\n",
+    "\n",
+    "mpioptions = \"-x NCCL_DEBUG=WARN -x SMDEBUG_LOG_LEVEL=ERROR \"\n",
+    "mpioptions += (\n",
+    "    \"-x SMP_DISABLE_D2D=1 -x SMP_D2D_GPU_BUFFER_SIZE_BYTES=1 -x SMP_NCCL_THROTTLE_LIMIT=1 \"\n",
+    ")\n",
+    "mpioptions += \"-x FI_EFA_USE_DEVICE_RDMA=1 -x FI_PROVIDER=efa -x RDMAV_FORK_SAFE=1\"\n",
+    "\n",
+    "metric_definitions = [\n",
+    "    {\"Name\": \"base_metric\", \"Regex\": \"<><><><><><>\"}\n",
+    "]  # Add your custom metric definitions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Specify a HuggingFace Dataset"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here you specify the dataset from HuggingFace that you want to train on. We provide two choices here (`wikitext` and `openwebtext`), but you can use any text dataset available for your runs."
```

Comment:
```suggestion
    "In this step, you specify the dataset from Hugging Face that you want to train on. We provide two choices here (`wikitext` and `openwebtext`). You can use any text dataset available for your runs."
```

Reply:
I added an additional note here, about the larger datasets taking longer to process.